log-manager/src/main/java/io/airlift/log/RollingFileHandler.java

```diff
@@ -0,0 +1,579 @@
+                catch (IOException e) {
+                    reportError("Unable to delete compress log file", e, GENERIC_FAILURE);
+                }
+                return;
+            }
+
+            Path compressedFile = originalFile.resolveSibling(originalFile.getFileName() + compressionExtension);
+
+            // 1. Move temp file to final compressed name
+            LogFileName compressedFileName = originalLogFileName.withCompression(compressedFile);
+            try {
+                Files.move(tempFile, compressedFile, ATOMIC_MOVE);
+            }
+            catch (IOException e) {
+                // add the original file back to the history manager
+                logHistoryManager.addFile(originalFile, originalLogFileName, originalFileSize);
+
+                // move failed, delete the temp file
+                try {
+                    Files.deleteIfExists(tempFile);
+                }
+                catch (IOException ignored) {
+                    // delete failed, system will attempt to delete temp files periodically
+                }
+            }
+            logHistoryManager.addFile(compressedFile, compressedFileName, compressedSize);
+
+            // 3. Delete original file
+            try {
+                Files.deleteIfExists(originalFile);
+            }
+            catch (IOException deleteException) {
+                // delete failed, try to move the file out of the way
+                try {
+                    Files.move(originalFile, originalFile.resolveSibling(DELETED_PREFIX + originalFile.getFileName()), ATOMIC_MOVE);
+                }
+                catch (IOException ignored) {
+                    // delete and move failed, there is nothing that can be done
+                    reportError("Unable to delete original file after compression", deleteException, GENERIC_FAILURE);
+                }
+            }
+        }
+    }
+
+    private static void tryCleanupTempFiles(Path masterLogFile)
+    {
+        try {
+            for (Path file : MoreFiles.listFiles(masterLogFile.getParent())) {
+                String fileName = file.getFileName().toString();
+                String fileNameWithoutPrefix;
+                if (fileName.startsWith(TEMP_PREFIX)) {
+                    fileNameWithoutPrefix = fileName.substring(TEMP_PREFIX.length());
+                }
+                else if (fileName.startsWith(DELETED_PREFIX)) {
+                    fileNameWithoutPrefix = fileName.substring(DELETED_PREFIX.length());
+                }
+                else {
+                    continue;
+                }
+                if (LogFileName.forName(masterLogFile.getFileName().toString(), fileNameWithoutPrefix).isPresent()) {
+                    // this is our temp or "to be deleted' file, so try to delete it
+                    try {
+                        Files.deleteIfExists(file);
+                    }
+                    catch (IOException ignored) {
+                    }
```

Comment:
Do we want to `reportError` here?

Reply:
I wasn't sure.  This method gets called every time we compress, so if there is a problem that gets fixed we cleanup.